.github/action_scripts/send_transactions/allow-spends-and-spend-transactions.js

```diff
@@ -417,6 +419,196 @@ const executeInvalidTransactionWorkflow = async (createTransactionFn, workflowNa
     }
 };
 
+const createInvalidSignatureTransaction = async (sourcePrivateKey, l0Url, l1Url, isCurrency = false) => {
+    const sourceAccount = createAndConnectAccount(sourcePrivateKey, { l0Url, l1Url }, isCurrency);
+    const ammAddress = CONSTANTS.CURRENCY_TOKEN_ID;
+
+    const allowSpend = await createAllowSpendTransaction(sourceAccount, ammAddress, l1Url, l0Url, isCurrency);
+    
+    const validProof = await generateProof(allowSpend, sourcePrivateKey, sourceAccount, SerializerType.BROTLI);
+    
+    const invalidProof = validProof.substring(0, validProof.length - 10) + 'INVALID' + validProof.substring(validProof.length - 5);
+    
+    logWorkflow.info(`Original proof: ${validProof.substring(0, 50)}...`);
+    logWorkflow.info(`Tampered proof: ${invalidProof.substring(0, 50)}...`);
+
+    try {
+        const body = { value: allowSpend, proofs: [invalidProof] };
+        logWorkflow.info(`Sending allow spend with invalid signature: ${JSON.stringify(body.value)}`);
+        const response = await axios.post(`${l1Url}/allow-spends`, body);
+        logWorkflow.success(`Response: ${JSON.stringify(response.data)}`);
+        return {
+            address: sourceAccount.address,
+            hash: response.data.hash,
+            amount: allowSpend.amount,
+            fee: allowSpend.fee,
+            lastValidEpochProgress: allowSpend.lastValidEpochProgress
+        };
+    } catch (error) {
+        if (error.response && error.response.status === 400) {
+            logWorkflow.success(`Transaction correctly rejected with 400 Bad Request: ${error.response.data}`);
+            return {
+                address: sourceAccount.address,
+                rejected: true,
+                error: error.response.data
+            };
+        }
+        logWorkflow.error('Error sending allow spend with invalid signature', error);
+        throw error;
+    }
+};
+
+const createDoubleSpendTransaction = async (sourcePrivateKey, dagL1Url, extendedDagL1Url, l0Url) => {
+    const sourceAccount = createAndConnectAccount(sourcePrivateKey, { l0Url, l1Url: dagL1Url }, false);
+    const ammAddress = CONSTANTS.CURRENCY_TOKEN_ID;
+
+    const allowSpend = await createAllowSpendTransaction(sourceAccount, ammAddress, dagL1Url, l0Url, false);
+    
+    const proof1 = await generateProof(allowSpend, sourcePrivateKey, sourceAccount, SerializerType.BROTLI);
+    const proof2 = await generateProof(allowSpend, sourcePrivateKey, sourceAccount, SerializerType.BROTLI);
+    
+    logWorkflow.info(`Created allow spend: ${JSON.stringify(allowSpend)}`);
+    logWorkflow.info(`Generated two proofs for the same allow spend`);
+
+    try {
+        const body1 = { value: allowSpend, proofs: [proof1] };
+        logWorkflow.info(`Sending allow spend to DAG L1: ${dagL1Url}`);
+        const response1 = await axios.post(`${dagL1Url}/allow-spends`, body1);
```

Comment:
const [response1, response2] = await Promise.all([request1, request2]);

Change this to only await so both requests are sent at the same time (better test of conflict detection if they're received at identical times.)

Reply:
done